src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,111 @@
 package core.basesyntax;
 
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private static final int GROW_CONSTANT = 2;
+
+    private List<Node<K, V>>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new List[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeeded();
+        int index = getIndex(key);
+        if (buckets[index] != null) {
+            Node<K, V> lastNode = null;
+            for (Node<K, V> node : buckets[index]) {
+                lastNode = node;
+                if (Objects.equals(node.key, key)) {
+                    node.value = value;
+                    return;
+                }
+            }
+            Node<K, V> newNode = createNewNode(key, value);
+            buckets[index].add(newNode);
+            if (lastNode != null) {
+                lastNode.next = newNode;
+            }
+        } else {
+            buckets[index] = new LinkedList<>();
+            buckets[index].add(createNewNode(key, value));
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        if (buckets[index] != null) {
+            for (Node<K, V> node : buckets[index]) {
+                if (Objects.equals(node.key, key)) {
+                    return node.value;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return this.size;
+    }
+
+    private void resizeIfNeeded() {
+        if ((float) size / buckets.length > DEFAULT_LOAD_FACTOR) {
+            resize();
```

Comment:
Love this idea of checking if an array is full, cool

Reply:
thx